### DISCUSSION TITLE
punctuation and tone

### DISCUSSION THREAD

**Discussion by ZYJGO** [yl4579/StyleTTS2#221] [Q&A]
I've been using Styletts2 lately since it allows me to kind of control the emotion of the generated speech using voice cloning, which is the best TTS for my use case, 

but one problem is that I get the exact same tone whether or not I add a question mark or exclamation point at the end of the sentence. I assume this is because it's not an auto regressive model, but is it possible to e.g. enhance the questioning tone of the generated sentences?

**Comment by meng2468** [comment]
Since the LibriTTS dataset doesn't include punctuation, the StyleTTS2 model won't change it's tone based on it.
If you have your own audio files and their transcriptions including the punctuation, the model should be able to learn it through training.

Maybe a quick-fix if that's too much effort: Have multiple reference files for different end of sentence punctuations?

**Reply by ZYJGO** [reply]
that's a good idea, thank you!

**Reply by tanishbajaj101** [reply]
> Since the LibriTTS dataset doesn't include punctuation, the StyleTTS2 model won't change it's tone based on it. If you have your own audio files and their transcriptions including the punctuation, the model should be able to learn it through training.
> 
> Maybe a quick-fix if that's too much effort: Have multiple reference files for different end of sentence punctuations?

when I used LJSpeech checkpoint, it gave me similar problem.
On downloading LJSpeech dataset, I found it already does have punctuations, a lot of them infact
